### ISSUE TITLE
Display device's Description field instead of hostname or ip

### ISSUE THREAD

**Issue by nicole212** [librenms/librenms#1000]
Librenms uses device's hostname field or ip as it's main field. This field standing for the device appears in all the device-related tables. what my suggestion is that we can use the Description field instead of hostnem or ip. It's because that we can set a device's description message with  a easy-to-understanding text  freely in setting sheet. I think this may make a better user experience. :)

**Comment by laf** [comment]
I'd be interested to hear from @paulgear and @f0o on this one but my take is we shouldn't do this, hostname and IP are both unique enough to describe a device (hostnames more so from a descriptive point of view), the work involved in swapping out the hostname for the description would be a huge pain and imho not worth it.

**Comment by nicole212** [comment]
Thanks for your reply. Hostname do be enough to describe a device . But I have to set each device's hostname through the  "/etc/hosts" . This operation is beyond Librenms and will be a heavy work when monitoring numerous devices. Can you give me some suggestion? Thanks again! :)

**Comment by laf** [comment]
But setting a description is also manual work for each device so I'm not sure it will save you any time, you'd be much better to give the devices better hostnames in the first place.

**Comment by f0o** [comment]
Seeing it like @laf 
No point in moving the manual work from /etc/hosts to the webui - workload will still be the same.

**Comment by paulgear** [comment]
Moving the manual work from /etc/hosts to the web UI would increase workload, because at least /etc/hosts changes can be automated.  But this issue does bring up a good point: we don't actually display the description in the device list or device overview screen (as far as I can tell).

**Comment by nicole212** [comment]
As what you said, workload is always there and can not be reduced by any way. May be my point is wanting to do this work on the web UI . If this job is worthless, I will go on  editing /etc/host. Thanks again@laf ,@f0o,@paulgear :)

**Comment by laf** [comment]
@nicole212 It's not worthless if it's of use to people but I think something like this just may not be. One concern I'd have if someone did contribute the code to enable it via config.php to see devices by description would be having to support that going forward as more and more pages / features are added that display the hostname by default.

If you wrapped it into a callable function then that would possibly work but over all I'm not a huge fan of the idea I'm afraid :/

**Comment by tomck** [comment]
I wish this weren't closed. We use LibreNMS's dashboard as a wall monitor device, and when we see a warning about an IP address like 10.0.1.251, we have to think "what is that!?"

For a while I thought this was our problem, but then I went to All Devices -> Network and saw that the names of the devices were right under their IP addresses! I tried to edit the columns of the Unacknowledged Alerts widget on the dashboard and there is NO option to show this instead of or in addition to the hostname.

So, is your solution that we need PTR records for IP to hostname for internal devices? Seems a little obtuse.
